From 1f001de516f44f3c7dd6f1315fdb54f138d62d36 Mon Sep 17 00:00:00 2001
From: jagaur <jagaur@habana.ai>
Date: Thu, 05 Jun 2025 12:34:16 +0300
Subject: [PATCH] [SW-230770] gaudi3: fix crc32 calculation

This patch adds a WA to fix crc32 calculation error which
occurs when the file path contains a hexadecimal string.

Issue was seen during gerrit to github migration where the
jenkins jobs being run were using a different folder structure
and that path contained a hexadecimal string.

Change-Id: Ieb65d80e441c9bc304b07de2c8c88d64c6a3202a
---

diff --git a/board/habanalabs/common/post-image-spi-arc_global.sh b/board/habanalabs/common/post-image-spi-arc_global.sh
index a567df6..543a7b3 100755
--- a/board/habanalabs/common/post-image-spi-arc_global.sh
+++ b/board/habanalabs/common/post-image-spi-arc_global.sh
@@ -139,7 +139,12 @@
 
 	# reverse CRC32 source as DMA engine works like that
 	hl_reverse_words ${FILE} ${FILE}.out 4
-	CRC32=$(crc32 ${FILE}.out)
+
+	# workround to avoid crc calculation error when absolute file path contains hex string
+	cp "${FILE}.out" /tmp/.
+	CRC32=$(crc32 /tmp/$(basename "${FILE}.out"))
+	rm -rf /tmp/$(basename "${FILE}.out")
+
 	echo ${CRC32} | xxd -r -p > ${FILE}.crc32
 }
 
